x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java

```diff
@@ -1199,26 +1196,19 @@ static void writeUnsupportedEsField(PlanStreamOutput out, UnsupportedEsField uns
 
     // -- BinaryComparison
 
-    static BinaryComparison readBinComparison(PlanStreamInput in, String name) throws IOException {
+    static EsqlBinaryComparison readBinComparison(PlanStreamInput in, String name) throws IOException {
         var source = in.readSource();
-        var operation = in.readEnum(BinaryComparisonProcessor.BinaryComparisonOperation.class);
+        EsqlBinaryComparison.BinaryComparisonOperation operation = EsqlBinaryComparison.BinaryComparisonOperation.readFromStream(in);
         var left = in.readExpression();
         var right = in.readExpression();
+        // TODO: Remove zoneId entirely
```

Comment:
Why this TODO?

Reply:
I mean, isn't there the possibility of having date > '2024-01-01T01:00:00` on a specific timezone (see Kibana) sometime in the future?

Reply:
The TODO is because it's a field we don't currently use or test; none of the actual ES|QL implementations actually use it, so it only looks like it does something, while actually doing nothing.  I find that confusing, since I tend to not expect to carry around unused fields.  In this particular case, you can see I am not even passing this value on to the constructor.

Beyond that, I don't see the reason that binary comparisons, and only binary comparisons, need to track the time zone they are operating in.  If a date is time zone aware, that information should be encoded in the date (`2024-04-25T09:15:00+06:00` or similar).  Carrying the timezone as part of the operator makes that very confusing.  What does it mean to say `date > 2024-04-25T09:15:00+06:00 in CET`?  And if that does have a meaning, why do we not also need it for (e.g.) subtraction of dates?

Ultimately, ES|QL doesn't have timezone support yet, and we don't know what that will look like. Maybe it will be the case that when building that, we decided we need to add back in a timezone here.  But until we have a plan for that, I don't see why we should carry around a confusing, unused value.